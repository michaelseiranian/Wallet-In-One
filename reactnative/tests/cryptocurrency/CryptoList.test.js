import CryptoList from "../../screens/cryptocurrency/CryptoList";
import {render, screen, act, waitFor, fireEvent} from "@testing-library/react-native";
import cryptoWallets from "../crypto_wallets/fixtures/cryptoWallets.json"

describe('<CryptoList />', () => {

  const params = {
    navigation: {
      navigate: jest.fn(),
      goBack: jest.fn()
    }
  };

  beforeEach(() => {
    global.fetch =  jest.fn( async () => {
      return Promise.resolve({
        status: 200,
      })
    })
  })

  afterEach(() => {
    global.fetch.mockClear();
    delete global.fetch;
  })

  it('snapshot test', async () => {
    const snapshot = render(<CryptoList {...params} />)

    // Test navigation
    await act(async () => {
      fireEvent.press(await screen.getByText('Insights'))
    })

    expect(snapshot).toMatchSnapshot()

  })

  it('display wallets and exchanges test', async () => {

    global.fetch =  jest.fn( async (api) => {
      const exchangeBalancesPattern = /get_exchange_balances/
      const walletPattern = /crypto_wallets/
      const exchangePattern = /crypto-exchanges/

      if (exchangeBalancesPattern.test(api)) {
        return Promise.resolve({
          status: 200, json: () => ([
            {
              "x": "Binance",
              "y": 0.13,
              "id": 2
            }
          ])
        })
      } else if (walletPattern.test(api)) {
        return Promise.resolve({
          status: 200, json: () => cryptoWallets
        })
      } else if (exchangePattern.test(api)) {
        return Promise.resolve({
          status: 200, json: () => cryptoWallets
        })
      }

    })
    const snapshot = render(<CryptoList {...params} />)

    await act(async () => {

    })

  })

  it('display empty wallets and empty exchanges test', async () => {
    const snapshot = render(<CryptoList {...params} />)

    await act(async () => {
      expect(screen.getByText("There are no wallets to display. Try connect a crypto wallet."))
      expect(screen.getByText("There are no exchanges to display. Try connect a crypto exchange."))
    })

  })

  it('update crypto wallet and exchange test', async () => {
    global.fetch =  jest.fn( async (api) => {
      const exchangeBalancesPattern = /get_exchange_balances/
      const walletPattern = /crypto_wallets/
      const exchangePattern = /crypto-exchanges/

      if (exchangeBalancesPattern.test(api)) {
        return Promise.resolve({
          status: 200, json: () => ([
            {
              "x": "Binance",
              "y": 0.13,
              "id": 2
            }
          ])
        })
      } else if (walletPattern.test(api)) {
        return Promise.resolve({
          status: 200, json: () => cryptoWallets
        })
      } else if (exchangePattern.test(api)) {
        return Promise.resolve({
          status: 200, json: () => cryptoWallets
        })
      }

    })

    const snapshot = render(<CryptoList {...params} />)

    await act(async () => {
      fireEvent.press(await screen.getByText('Update'))
    })

    expect(snapshot).toMatchSnapshot()

  })

  it('update invalid crypto wallet and exchange test', async () => {
    global.fetch =  jest.fn( async (api) => {
      const exchangeBalancesPattern = /get_exchange_balances/
      const walletPattern = /crypto_wallets/
      const exchangePattern = /crypto-exchanges/

      if (exchangeBalancesPattern.test(api)) {
        return Promise.resolve({
          status: 200, json: () => ([
            {
              "x": "Binance",
              "y": 0.13,
              "id": 2
            }
          ])
        })
      } else if (walletPattern.test(api)) {
        return Promise.resolve({
          status: 400, json: () => cryptoWallets
        })
      } else if (exchangePattern.test(api)) {
        return Promise.resolve({
          status: 400, json: () => cryptoWallets
        })
      }

    })

    const snapshot = render(<CryptoList {...params} />)

    await act(async () => {
      fireEvent.press(await screen.getByText('Update'))
    })

    expect(snapshot).toMatchSnapshot()

  })

  it('update invalid crypto wallet and exchange test', async () => {
    global.fetch =  jest.fn( async (api) => {
      const exchangeBalancesPattern = /get_exchange_balances/
      const walletPattern = /crypto_wallets/
      const exchangePattern = /crypto-exchanges/

      if (exchangeBalancesPattern.test(api)) {
        return Promise.resolve({
          status: 200, json: () => ([
            {
              "x": "Binance",
              "y": 0.13,
              "id": 2
            }
          ])
        })
      } else if (walletPattern.test(api)) {
        return Promise.reject()
      } else if (exchangePattern.test(api)) {
        return Promise.resolve({
          status: 400, json: () => Promise.reject()
        })
      }

    })

    const snapshot = render(<CryptoList {...params} />)

    await act(async () => {
      fireEvent.press(await screen.getByText('Update'))
    })

  })

})