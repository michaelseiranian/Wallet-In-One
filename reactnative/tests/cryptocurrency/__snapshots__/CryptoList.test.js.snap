// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CryptoList /> snapshot test 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignSelf": "center",
              "backgroundColor": "purple",
              "borderRadius": 10,
              "marginHorizontal": 8,
              "marginVertical": 4,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Insights
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignSelf": "center",
              "backgroundColor": "purple",
              "borderRadius": 10,
              "marginHorizontal": 8,
              "marginVertical": 4,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Update
          </Text>
        </View>
      </View>
      <Text
        style={
          [
            {
              "alignSelf": "center",
              "fontSize": 30,
              "fontWeight": "900",
            },
            {
              "color": "black",
              "marginTop": 10,
            },
          ]
        }
      >
        Wallets
      </Text>
      <View
        style={
          [
            {
              "marginBottom": 30,
              "marginHorizontal": 10,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "black",
            }
          }
        >
          There are no wallets to display. Try connect a crypto wallet.
        </Text>
      </View>
      <Text
        style={
          [
            {
              "alignSelf": "center",
              "fontSize": 30,
              "fontWeight": "900",
            },
            {
              "color": "black",
            },
          ]
        }
      >
        Exchanges
      </Text>
      <View
        style={
          [
            {
              "marginBottom": 30,
              "marginHorizontal": 10,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "black",
            }
          }
        >
          There are no exchanges to display. Try connect a crypto exchange.
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`<CryptoList /> update crypto wallet and exchange test 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignSelf": "center",
              "backgroundColor": "purple",
              "borderRadius": 10,
              "marginHorizontal": 8,
              "marginVertical": 4,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Insights
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignSelf": "center",
              "backgroundColor": "purple",
              "borderRadius": 10,
              "marginHorizontal": 8,
              "marginVertical": 4,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Update
          </Text>
        </View>
      </View>
      <Text
        style={
          [
            {
              "alignSelf": "center",
              "fontSize": 30,
              "fontWeight": "900",
            },
            {
              "color": "black",
              "marginTop": 10,
            },
          ]
        }
      >
        Wallets
      </Text>
      <View
        style={
          [
            {
              "marginBottom": 30,
              "marginHorizontal": 10,
            },
          ]
        }
      >
        <View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "borderRadius": 10,
                    "flexDirection": "row",
                    "marginVertical": 5,
                    "padding": 10,
                  },
                  {
                    "backgroundColor": "purple",
                  },
                ]
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "paddingRight": 10,
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    {
                      "height": 30,
                      "width": 30,
                    }
                  }
                />
              </View>
              <View
                style={
                  {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  style={{}}
                >
                  <Text
                    style={
                      {
                        "color": "black",
                        "fontSize": 25,
                        "fontWeight": "700",
                      }
                    }
                  >
                    Bitcoin
                  </Text>
                  <Text
                    style={
                      [
                        {
                          "flex": 1,
                          "fontWeight": "700",
                        },
                        {
                          "color": "black",
                        },
                      ]
                    }
                  >
                    0.01344898
                     
                    BTC
                  </Text>
                  <Text
                    style={
                      [
                        {
                          "flex": 1,
                          "fontWeight": "700",
                        },
                        {
                          "color": "black",
                        },
                      ]
                    }
                  >
                    £
                    0.00
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "borderRadius": 10,
                    "flexDirection": "row",
                    "marginVertical": 5,
                    "padding": 10,
                  },
                  {
                    "backgroundColor": "purple",
                  },
                ]
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "paddingRight": 10,
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    {
                      "height": 30,
                      "width": 30,
                    }
                  }
                />
              </View>
              <View
                style={
                  {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  style={{}}
                >
                  <Text
                    style={
                      {
                        "color": "black",
                        "fontSize": 25,
                        "fontWeight": "700",
                      }
                    }
                  >
                    Dash
                  </Text>
                  <Text
                    style={
                      [
                        {
                          "flex": 1,
                          "fontWeight": "700",
                        },
                        {
                          "color": "black",
                        },
                      ]
                    }
                  >
                    1.33461905
                     
                    DASH
                  </Text>
                  <Text
                    style={
                      [
                        {
                          "flex": 1,
                          "fontWeight": "700",
                        },
                        {
                          "color": "black",
                        },
                      ]
                    }
                  >
                    £
                    0.00
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          [
            {
              "alignSelf": "center",
              "fontSize": 30,
              "fontWeight": "900",
            },
            {
              "color": "black",
            },
          ]
        }
      >
        Exchanges
      </Text>
      <View
        style={
          [
            {
              "marginBottom": 30,
              "marginHorizontal": 10,
            },
          ]
        }
      >
        <View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "borderRadius": 10,
                  "flexDirection": "row",
                  "marginVertical": 5,
                  "padding": 10,
                },
                {
                  "backgroundColor": "purple",
                },
              ]
            }
            testID="exchangeAsset"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "paddingRight": 10,
                }
              }
            >
              <Image
                style={
                  {
                    "height": 30,
                    "width": 30,
                  }
                }
              />
            </View>
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={{}}
              >
                <Text
                  style={
                    {
                      "color": "black",
                      "fontSize": 25,
                      "fontWeight": "700",
                    }
                  }
                />
                <Text
                  style={
                    [
                      {
                        "flex": 1,
                        "fontWeight": "700",
                      },
                      {
                        "color": "black",
                      },
                    ]
                  }
                >
                  Total Balance: £
                  0.00
                </Text>
              </View>
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "borderRadius": 10,
                  "flexDirection": "row",
                  "marginVertical": 5,
                  "padding": 10,
                },
                {
                  "backgroundColor": "purple",
                },
              ]
            }
            testID="exchangeAsset"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "paddingRight": 10,
                }
              }
            >
              <Image
                style={
                  {
                    "height": 30,
                    "width": 30,
                  }
                }
              />
            </View>
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={{}}
              >
                <Text
                  style={
                    {
                      "color": "black",
                      "fontSize": 25,
                      "fontWeight": "700",
                    }
                  }
                />
                <Text
                  style={
                    [
                      {
                        "flex": 1,
                        "fontWeight": "700",
                      },
                      {
                        "color": "black",
                      },
                    ]
                  }
                >
                  Total Balance: £
                  0.13
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`<CryptoList /> update invalid crypto wallet and exchange test 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignSelf": "center",
              "backgroundColor": "purple",
              "borderRadius": 10,
              "marginHorizontal": 8,
              "marginVertical": 4,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Insights
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignSelf": "center",
              "backgroundColor": "purple",
              "borderRadius": 10,
              "marginHorizontal": 8,
              "marginVertical": 4,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Update
          </Text>
        </View>
      </View>
      <Text
        style={
          [
            {
              "alignSelf": "center",
              "fontSize": 30,
              "fontWeight": "900",
            },
            {
              "color": "black",
              "marginTop": 10,
            },
          ]
        }
      >
        Wallets
      </Text>
      <View
        style={
          [
            {
              "marginBottom": 30,
              "marginHorizontal": 10,
            },
          ]
        }
      >
        <View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "borderRadius": 10,
                    "flexDirection": "row",
                    "marginVertical": 5,
                    "padding": 10,
                  },
                  {
                    "backgroundColor": "purple",
                  },
                ]
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "paddingRight": 10,
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    {
                      "height": 30,
                      "width": 30,
                    }
                  }
                />
              </View>
              <View
                style={
                  {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  style={{}}
                >
                  <Text
                    style={
                      {
                        "color": "black",
                        "fontSize": 25,
                        "fontWeight": "700",
                      }
                    }
                  >
                    Bitcoin
                  </Text>
                  <Text
                    style={
                      [
                        {
                          "flex": 1,
                          "fontWeight": "700",
                        },
                        {
                          "color": "black",
                        },
                      ]
                    }
                  >
                    0.01344898
                     
                    BTC
                  </Text>
                  <Text
                    style={
                      [
                        {
                          "flex": 1,
                          "fontWeight": "700",
                        },
                        {
                          "color": "black",
                        },
                      ]
                    }
                  >
                    £
                    0.00
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "borderRadius": 10,
                    "flexDirection": "row",
                    "marginVertical": 5,
                    "padding": 10,
                  },
                  {
                    "backgroundColor": "purple",
                  },
                ]
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "paddingRight": 10,
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    {
                      "height": 30,
                      "width": 30,
                    }
                  }
                />
              </View>
              <View
                style={
                  {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  style={{}}
                >
                  <Text
                    style={
                      {
                        "color": "black",
                        "fontSize": 25,
                        "fontWeight": "700",
                      }
                    }
                  >
                    Dash
                  </Text>
                  <Text
                    style={
                      [
                        {
                          "flex": 1,
                          "fontWeight": "700",
                        },
                        {
                          "color": "black",
                        },
                      ]
                    }
                  >
                    1.33461905
                     
                    DASH
                  </Text>
                  <Text
                    style={
                      [
                        {
                          "flex": 1,
                          "fontWeight": "700",
                        },
                        {
                          "color": "black",
                        },
                      ]
                    }
                  >
                    £
                    0.00
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          [
            {
              "alignSelf": "center",
              "fontSize": 30,
              "fontWeight": "900",
            },
            {
              "color": "black",
            },
          ]
        }
      >
        Exchanges
      </Text>
      <View
        style={
          [
            {
              "marginBottom": 30,
              "marginHorizontal": 10,
            },
          ]
        }
      >
        <View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "borderRadius": 10,
                  "flexDirection": "row",
                  "marginVertical": 5,
                  "padding": 10,
                },
                {
                  "backgroundColor": "purple",
                },
              ]
            }
            testID="exchangeAsset"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "paddingRight": 10,
                }
              }
            >
              <Image
                style={
                  {
                    "height": 30,
                    "width": 30,
                  }
                }
              />
            </View>
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={{}}
              >
                <Text
                  style={
                    {
                      "color": "black",
                      "fontSize": 25,
                      "fontWeight": "700",
                    }
                  }
                />
                <Text
                  style={
                    [
                      {
                        "flex": 1,
                        "fontWeight": "700",
                      },
                      {
                        "color": "black",
                      },
                    ]
                  }
                >
                  Total Balance: £
                  0.00
                </Text>
              </View>
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "borderRadius": 10,
                  "flexDirection": "row",
                  "marginVertical": 5,
                  "padding": 10,
                },
                {
                  "backgroundColor": "purple",
                },
              ]
            }
            testID="exchangeAsset"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "paddingRight": 10,
                }
              }
            >
              <Image
                style={
                  {
                    "height": 30,
                    "width": 30,
                  }
                }
              />
            </View>
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={{}}
              >
                <Text
                  style={
                    {
                      "color": "black",
                      "fontSize": 25,
                      "fontWeight": "700",
                    }
                  }
                />
                <Text
                  style={
                    [
                      {
                        "flex": 1,
                        "fontWeight": "700",
                      },
                      {
                        "color": "black",
                      },
                    ]
                  }
                >
                  Total Balance: £
                  0.13
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
